### PR TITLE
Avoid wait_idle calls in script_run

### PR DESCRIPTION
**Pull request by coolo**
Replaced it with wait_serial for a random string

### CODE REVIEW

File: testapi.pm [hunk 1]
```diff
@@ -294,6 +294,10 @@ sub wait_idle {
     my $timeout = shift || $bmwqemu::idle_timeout;
     $timeout = bmwqemu::scale_timeout($timeout);
 
+    # report wait_idle calls more offensiv to help with https://progress.opensuse.org/issues/5830
```
Comment (okurz) [inline testapi.pm]: s/offensiv/offensively/

**Comment by okurz** [comment]
-1, fix typo and put a bit more details in the commit message for reasoning and benefit of this change. You might want to reference the progress issue. After these I would be ok if you merge it directly, i.e. --> auto-LGTM

File: distribution.pm [hunk 2]
```diff
@@ -101,8 +102,15 @@ sub script_run {
     # start console application
     my ($self, $name, $wait) = @_;
 
-    testapi::type_string "$name\n";
-    testapi::wait_idle($wait);
+    testapi::type_string "$name";
+    if ($wait > 0) {
```
Comment (aaannz) [inline distribution.pm]: Please add check if `$wait` is defined. Else we will have undef warnings all over the place.

Comment (aaannz) [inline distribution.pm]: Ah, it's added on higher level. Ok, nevermind.

File: testapi.pm [hunk 3]
```diff
@@ -294,6 +294,10 @@ sub wait_idle {
...
+    use Carp qw(cluck);
+    cluck "Wait_idle called";
```
Comment (okurz) [inline testapi.pm]: I would add "deprecated: …" here

**Comment by aaannz** [comment]
My one concern is use case like in `installation/zdup.pm`. `script_run "zypper dup"`, then `wait_serial` to check output. It's called without explicit `$timeout` so it will get default 30s. Would new `script_run` "steal" serial output? I think so.

File: testapi.pm (hunk 1 above)
Comment (coolo) [inline testapi.pm]: dude!

File: testapi.pm (hunk 3 above)
Comment (coolo) [inline testapi.pm]: it's not deprecated. I just want to reduce it's usage

**Comment by coolo** [comment]
yes, it will - that's why I added , 0 to script_runs waiting itself.

**Comment by coolo** [comment]
the API change is a big concern to me too - but I disagree with adding another function. The overhead in documentation and testing is just too big and we need to do this cleanup to be portable.

Fedora uses stable openQA - and we'll need an upgrade guide.

**Comment by coolo** [comment]
Of course if you can think of a gentle way to do the switch, I consider it. But remember we have assert_script_sudo, script_sudo, script_run and assert_script_run all sharing the same problem.

File: testapi.pm (hunk 3 above)
Comment (okurz) [inline testapi.pm]: IMHO it's deprecated but not obsolete. "deprecated" means "please don't use" as in "reduce it's usage". That's what you said.

File: testapi.pm (hunk 1 above)
Comment (okurz) [inline testapi.pm]: well, if you change other stuff then you can also change this :-)

**Comment by okurz** [comment]
> Fedora uses stable openQA - and we'll need an upgrade guide.

Then a first easy step is to mark the "wait_idle" as deprecated with the corresponding annoying log warning in "stable openQA"

**Comment by coolo** [comment]
that won't help a dime as long as script_run and send_key use it. It will be just *very* annoying

**Comment by sysrich** [comment]
I agree with @coolo

The first step to either deprecation or obsolescence is removing wait_idle
from script_run and send_key

I don't care which term or approach we take, as long as the outcome is that
we can move away from wait_idle usage in our test suite forthwith

And that means removing its use from those functions now,despite it being a
major change which will necessitate some work for us and other openQA users

**Comment by okurz** [comment]
I am fine if we go that road but I don't agree that "the first step to deprecation" should be changing existing functionality. That is how deprecation differs from plain changing behaviour. I see marking "wait_idle" as deprecated as a hint to other users of openQA. For os-autoinst-distri-opensuse developers we can of course try to get rid of wait_idle but that is not directly related.

**Comment by coolo** [comment]
I never said that wait_idle is obsolete or deprecated. You said that - I said, I want to get rid of its usage in our tests. Direct or indirect

**Comment by coolo** [comment]
extended the API documentationa and the commit log -  7e44b20

**Comment by okurz** [comment]
LGTM

**Comment by coolo** [comment]
testing the changes in staging atm: https://tortuga.suse.de/tests/16

**Comment by coolo** [comment]
(ran out of worker disk space)

**Comment by coolo** [comment]
merging this now so we can deploy it tomorrow morning to find more issues in the tests

**Pull request merged**